--- boost/config/compiler/visualc.hpp	2015-09-15 14:29:28.000000000 -0500
+++ boost/config/compiler/visualc.hpp.new	2015-09-15 14:30:06.000000000 -0500
@@ -320,7 +320,7 @@

 //
 // last known and checked version is 19.10.25017 (VC++ 2017):
-#if (_MSC_VER > 1910)
+#if (_MSC_VER > 2000)
 #  if defined(BOOST_ASSERT_CONFIG)
 #     error "Unknown compiler version - please run the configure tests and report the results"
 #  else
